debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/J9BCUtil.java

```diff
@@ -1151,52 +1153,67 @@ private static void dumpStackMapTable(PrintStream out, J9ROMClassPointer romclas
 
 			out.println("\n  StackMapTable\n    Stackmaps(" + stackMapCount.intValue() + "):");
 
-			for (int i = 0; i < stackMapCount.intValue(); i++) {
-				mapPC++;
+			boolean dumpBaseFrame = false;
+			int i = 0;
+			while (i < stackMapCount.intValue()) {
 				mapType = stackMapData.at(0).longValue();
 				stackMapData = stackMapData.add(1);
 
+				/* print indent */
+				if (dumpBaseFrame) {
+					dumpBaseFrame = false;
+					out.print("        base: ");
+				} else {
+					mapPC += 1;
+					out.print("      ");
+				}
+
 				if (mapType < 64) {
 					mapPC += mapType;
-					out.println("      pc: " + mapPC +" same");
+					out.println("pc: " + mapPC +" same");
 				} else if (mapType < 128) {
 					mapPC += (mapType - 64);
-					out.print("      pc: " + mapPC +" same_locals_1_stack_item: ");
+					out.print("pc: " + mapPC +" same_locals_1_stack_item: ");
 					stackMapData = dumpStackMapSlots(out, romclass, stackMapData, 1);
 					out.println();
-				} else if (mapType < 247) {
-					out.println("      UNKNOWN FRAME TAG: (" + mapType + ")\n");
+				} else if (mapType < CFR_STACKMAP_EARLY_LARVAL) {
+					out.format("UNKNOWN FRAME TAG: (" + mapType + ")%n%n");
+				} else if (mapType == CFR_STACKMAP_EARLY_LARVAL) {
```

Comment:
Do we need a class version check ? As for Pre-Valhalla classes, it should still treated as `UNKNOWN FRAME TAG`